### PR TITLE
Inventory HUD stuff

### PR DESCRIPTION
**Pull request by agentflare**

### CODE REVIEW

File: Live/Classes/HUD/HUD.cpp [hunk 1]
```diff
@@ -28,6 +28,10 @@ void HUD::initComponents() {
         new HUDBar("Stamina", 100, 50,
                    cocos2d::Director::getInstance()->getVisibleSize().height - 150);
     addChild(m_staminaBar, 2);
+
+    m_inventory =
+        new HUDInventory();
```
Comment (sadmansk) [inline Live/Classes/HUD/HUD.cpp]: put it on the same line.

File: Live/Classes/HUD/HUD.h [hunk 2]
```diff
@@ -2,7 +2,9 @@
 
 #include "cocos2d.h"
 #include "HUDBar.h"
+#include "HUDInventory.h"
 #include "Player.h"
+#include "HUDInventory.h"
```
Comment (sadmansk) [inline Live/Classes/HUD/HUD.h]: repeated includes...

File: Live/Classes/HUD/HUDInventory.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  for(int i = 0; i < INVENTORY_SLOTS; i++) {
+    HUDInventoryItem* item = new HUDInventoryItem( 5 + (5 + INVENTORY_SLOT_SIZE)*i, 5 );
+    m_items[i] = item;
+    // positioning stuff
+    addChild(item, i);
+  }
+}
+
+void HUDInventory::update() {
+  for (int i = 0; i < INVENTORY_SLOTS; i++) {
```
Comment (sadmansk) [inline Live/Classes/HUD/HUDInventory.cpp]: change to four space indentation over the whole file.

File: Live/Classes/HUD/HUDInventory.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+USING_NS_CC;
+
+HUDInventory::~HUDInventory() {
+
+}
+
+HUDInventory::HUDInventory() {
+  LayerColor::initWithColor(Color4B(255, 255, 255, 255));
+  double inv_width = 5+(INVENTORY_SLOT_SIZE+5)*INVENTORY_SLOTS;
```
Comment (sadmansk) [inline Live/Classes/HUD/HUDInventory.cpp]: What does `5` represent? put that in a macro with an appropriate name.

File: Live/Classes/HUD/HUDInventory.cpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  setPosition(Vec2(Director::getInstance()->getVisibleSize().width/2 - inv_width/2,0));
+  setContentSize(Size(inv_width,INVENTORY_SLOT_SIZE+10));
+
+  for(int i = 0; i < INVENTORY_SLOTS; i++) {
+    HUDInventoryItem* item = new HUDInventoryItem( 5 + (5 + INVENTORY_SLOT_SIZE)*i, 5 );
```
Comment (sadmansk) [inline Live/Classes/HUD/HUDInventory.cpp]: again, change `5`s here to an appropriate macro, also, no need to use an extra variable, just do `m_items[i] = new HUDInventoryItem(...);`

File: Live/Classes/MainScene.h [hunk 6]
```diff
@@ -1,6 +1,9 @@
 #ifndef __MAIN_SCENE_H__
 #define __MAIN_SCENE_H__
 
+#include "cocos2d.h"
+#include "Player.h"
+#include "HUD/HUD.h"
```
Comment (sadmansk) [inline Live/Classes/MainScene.h]: repeated includes again. Please do a review yourself first before making the PR.

**Review by sadmansk — CHANGES_REQUESTED** [review]

**Comment by agentflare** [comment]
@sadmansk on your review again, we merged in the HUD improvement stuff. pls review those also <3

File: Live/Classes/HUD/HUDInventory.cpp (hunk 5 above)
Comment (Jacqueuela) [reply Live/Classes/HUD/HUDInventory.cpp]: gotchu

File: Live/Classes/HUD/HUDInventory.cpp (hunk 3 above)
Comment (Jacqueuela) [reply Live/Classes/HUD/HUDInventory.cpp]: gotchu

**Review by sadmansk — APPROVED** [review]

**Pull request merged**